sources/api/schnauzer/src/helpers.rs

```diff
@@ -2080,6 +2117,86 @@ mod test_default {
     }
 }
 
+#[cfg(test)]
+mod test_if_not_null {
+    use super::*;
+    use handlebars::RenderError;
+    use serde::Serialize;
+    use serde_json::json;
+
+    // A thin wrapper around the handlebars render_template method that includes
+    // setup and registration of helpers
+    fn setup_and_render_template<T>(tmpl: &str, data: &T) -> Result<String, RenderError>
+    where
+        T: Serialize,
+    {
+        let mut registry = Handlebars::new();
+        registry.register_helper("if-not-null", Box::new(IfNotNullHelper));
+
+        registry.render_template(tmpl, data)
+    }
+
+    #[test]
+    fn null_value() {
+        let result = setup_and_render_template(
+            "{{#if-not-null setting}}foo{{/if-not-null}}",
```

Comment:
Is it possible to create this so that the natural/built-in conditionals work as expected, i.e. the native `#if`?

```suggestion
            "{{#if not_null setting}}foo{{/if}}",
```

Reply:
I couldn't get this to work. for some reason handlebars treats a boolean return from a helper as a string, so the `#if` helper resolves to `true` when we return `false` from `not_null`.

it could work if I returned an empty string on false, because handlebars treats empty string as falsy, but it felt hackier to me and essentially limits us to using it in the same scenarios as `#if-not-null`

Reply:
> I couldn't get this to work. for some reason handlebars treats a boolean return from a helper as a string

That's because the value returned by the helpers is a string, not a boolean. I had the same issue.